From a3fa0f94132a66cb9bae09fc7b53f9ce41147aef Mon Sep 17 00:00:00 2001
From: Firelight Flagboy <firelight.flagboy@gmail.com>
Date: Sun, 18 Feb 2024 14:17:01 +0100
Subject: [PATCH] Normalize the config loading

Prevent duplicate keys in the config by using the extra_config as the base
and fallback to the data_conf in case of falsy value.
---
 parsec/core/config.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/parsec/core/config.py b/parsec/core/config.py
index 491c019b9..893ba48b0 100644
--- a/parsec/core/config.py
+++ b/parsec/core/config.py
@@ -259,7 +259,11 @@ def load_config(config_dir: Path | None = None, **extra_config: Any) -> CoreConf
     if data_conf.get("gui_last_version"):
         data_conf["gui_last_version"] = data_conf["gui_last_version"].lstrip("v")
 
-    return config_factory(config_dir=config_dir, **data_conf, **extra_config, environ=os.environ)
+    data = {
+        k: extra_config.get(k) or data_conf.get(k)
+        for k in data_conf.keys() | extra_config.keys()
+    }
+    return config_factory(config_dir=config_dir, **data, environ=os.environ)
 
 
 def save_config(config: CoreConfig) -> None:
-- 
2.42.0

